lib/house.rb

```diff
@@ -0,0 +1,33 @@
+class House
+  attr_reader :price,
+              :address,
+              :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    false
+  end
+
+  def rooms_from_category(category)
+    bedrooms = @rooms.find_all do |room|
+      if room.category == category
+        room
+      end
+    end
+  end
```

Comment:
This is working, but the correct way to write this method would be like this:
```ruby
  def rooms_from_category(category)
    @rooms.find_all do |room|
      room.category == category
    end
  end
```

Reply:
After our enums II class, this makes a lot more sense! And since our test is looking for the rooms that return with a category of bedroom AND also with a category of basement, it does not make sense to have the code be assigned to bedrooms. Thank you so much on the clarification of this Megan! It really helped me to understand!